File: OdevRehber/app/src/main/java/com/xzcorp/odevrehber/MainActivity.java
package com.xzcorp.odevrehber;

import android.Manifest;
import android.annotation.SuppressLint;
import android.content.pm.PackageManager;
import android.database.Cursor;
import android.net.Uri;
import android.os.Bundle;
import android.provider.ContactsContract;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.ActivityCompat;
import androidx.core.content.ContextCompat;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    RecyclerView recyclerView;
    ArrayList<ContactModel> contactModels;
    ContactAdapter adapter;
    private int PERMISSION_CODE=102;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
        contactModels=new ArrayList<>();
        checkRuntimePermission();
    }

    private void checkRuntimePermission() {
        if(ContextCompat.checkSelfPermission(MainActivity.this, Manifest.permission.READ_CONTACTS)!=PackageManager.PERMISSION_GRANTED){
            ActivityCompat.requestPermissions(MainActivity.this,new String[]{Manifest.permission.READ_CONTACTS},PERMISSION_CODE);
        }
        else
            getContactList();
    }

    private void getContactList() {
        Uri uri= ContactsContract.Contacts.CONTENT_URI;
        String sort=ContactsContract.CommonDataKinds.Phone.DISPLAY_NAME+" ASC";
        Cursor cursor=getContentResolver().query(uri,null,null,null,sort);

        if(cursor.getCount()>0){
            while(cursor.moveToNext()){
                @SuppressLint("Range") String id=cursor.getString(cursor.getColumnIndex(ContactsContract.Contacts._ID));
                @SuppressLint("Range") String name=cursor.getString(cursor.getColumnIndex(ContactsContract.Contacts.DISPLAY_NAME));

                Uri phone_url=ContactsContract.CommonDataKinds.Phone.CONTENT_URI;
                String selection=ContactsContract.CommonDataKinds.Phone.CONTACT_ID+" =?";
                Cursor phone_cursor=getContentResolver().query(phone_url,null,selection,new String[]{id},null);

                if(phone_cursor.moveToNext()){
                    @SuppressLint("Range") String number=phone_cursor.getString(phone_cursor.getColumnIndex(ContactsContract.CommonDataKinds.Phone.NUMBER));
                    ContactModel model=new ContactModel();
                    model.setName(name);
                    model.setNumber(number);
                    contactModels.add(model);
                    phone_cursor.close();
                }
            }
            cursor.close();
        }
        recyclerView=findViewById(R.id.m_recyclerView);
        recyclerView.setHasFixedSize(true);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));

        adapter=new ContactAdapter(this,contactModels);
        recyclerView.setAdapter(adapter);
    }

    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
        if(requestCode==PERMISSION_CODE && grantResults.length>0 && grantResults[0]==PackageManager.PERMISSION_GRANTED){
            getContactList();
        }
        else {
            Toast.makeText(this,"İzin Gerekli",Toast.LENGTH_SHORT).show();
            checkRuntimePermission();
        }
    }
}

File: OdevUlkeMars/app/src/main/java/com/xzcorp/odevulkemars/MainActivity.kt
package com.xzcorp.odevulkemars

import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity() {
    lateinit var recyclerView: RecyclerView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
        recyclerView = findViewById(R.id.recyclerView)
        recyclerView.layoutManager=LinearLayoutManager(this, LinearLayoutManager.VERTICAL,false)
        recyclerView.adapter=Adapter(getModels())

    }

    fun getModels():MutableList<CountryModel>{
        val models=mutableListOf(CountryModel(R.drawable.azerbaycan,"Azerbaycan","Bakü", "azerbaycan.mp3"),
            CountryModel(R.drawable.belcika,"Belçika","Brüksel","belcika.mp3"),
            CountryModel(R.drawable.dijibouti,"Dijibouti","Dijibouti","dijibouti.mp3"),
            CountryModel(R.drawable.ispanya,"İspanya","Madrid","ispanya.mp3"),
            CountryModel(R.drawable.turkiye,"Türkiye","Ankara","turkiye.mp3"),
            CountryModel(R.drawable.japan,"Japonya","Tokyo","japonya.mp3"),
            CountryModel(R.drawable.china,"Çin","Pekin","cin.mp3"),
            CountryModel(R.drawable.greece,"Yunanistan","Atina","yunanistan.mp3"),
            CountryModel(R.drawable.uk,"İngiltere","London","ingiltere.mp3"),
            CountryModel(R.drawable.abd,"Amerika","Washington, DC","amerika.mp3"),
            CountryModel(R.drawable.uruguay,"Uruguay","Montevideo","uruguay.mp3"))
        return models
    }
}